server/src/main/java/io/unitycatalog/server/service/credential/CredentialOperations.java

```diff
@@ -0,0 +1,108 @@
+package io.unitycatalog.server.service.credential;
+
+import com.google.auth.oauth2.AccessToken;
+import io.unitycatalog.server.exception.BaseException;
+import io.unitycatalog.server.exception.ErrorCode;
+import io.unitycatalog.server.model.AwsCredentials;
+import io.unitycatalog.server.model.AzureUserDelegationSAS;
+import io.unitycatalog.server.model.GcpOauthToken;
+import io.unitycatalog.server.model.GenerateTemporaryTableCredentialResponse;
+import io.unitycatalog.server.model.GenerateTemporaryVolumeCredentialResponse;
+import io.unitycatalog.server.model.TableInfo;
+import io.unitycatalog.server.model.VolumeInfo;
+import io.unitycatalog.server.service.credential.aws.AwsCredentialVendor;
+import io.unitycatalog.server.service.credential.azure.AzureCredential;
+import io.unitycatalog.server.service.credential.azure.AzureCredentialVendor;
+import io.unitycatalog.server.service.credential.gcp.GcpCredentialVendor;
+import software.amazon.awssdk.services.sts.model.Credentials;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Set;
+
+import static io.unitycatalog.server.utils.Constants.URI_SCHEME_ABFS;
+import static io.unitycatalog.server.utils.Constants.URI_SCHEME_ABFSS;
+import static io.unitycatalog.server.utils.Constants.URI_SCHEME_GS;
+import static io.unitycatalog.server.utils.Constants.URI_SCHEME_S3;
+
+public class CredentialOperations {
+
+  private final AwsCredentialVendor awsCredentialVendor;
+  private final AzureCredentialVendor azureCredentialVendor;
+  private final GcpCredentialVendor gcpCredentialVendor;
+
+  public CredentialOperations() {
+    this.awsCredentialVendor = new AwsCredentialVendor();
+    this.azureCredentialVendor = new AzureCredentialVendor();
+    this.gcpCredentialVendor = new GcpCredentialVendor();
+  }
+
+  public GenerateTemporaryTableCredentialResponse vendCredentialForTable(TableInfo table) {
+    String tableStorageLocation = table.getStorageLocation();
+    if (tableStorageLocation == null || tableStorageLocation.isEmpty()) {
+      throw new BaseException(ErrorCode.FAILED_PRECONDITION, "Table storage location not found.");
+    }
+
+    URI storageLocationUri = URI.create(tableStorageLocation);
+
+    // FIXME!! Update privileges when access controls are implemented
+    CredentialContext credentialContext = CredentialContext.create(storageLocationUri,
+      Set.of(CredentialContext.Privilege.SELECT, CredentialContext.Privilege.UPDATE));
```

Comment:
Shouldn't the `TableOperation` in the `GenerateTemporaryTableCredentials` request be used here? Otherwise, this always returns a read-write credential even when the client requested a read credential.

(as pointed out in the comment, when authz is added privileges should be checked to make sure the client is authorized to perform the requested `TableOperation`)

Reply:
I can switch this to use the table operations directly here. There will probably be some translation of access control privileges and table operations anyways though. 

> always returns a read-write credential even when the client requested a read credential.
I think we just come from a little bit different approaches here where when someone asks for table creds, we just look up what privileges that user/role/etc. has on the table and give back a cred with those privs instead of having the client tell the server what privs it wants/needs.

Reply:
So, instead here, I added code to translate the Table (and Volume) operation to access control privileges in the generate cred handlers (so stuff coming from that path will use the operation passed in). Personally, think this is better instead of requiring areas (like the iceberg rest side and credential stuff) to use the unity model. We have to look up privileges anyways and translate to verify access. We could always change this later if we need to.